### PR TITLE
Reporting usage statistics by FE

### PR DESCRIPTION
**Pull request by gadomsky**

### CODE REVIEW

File: e2e-tests/docker-compose.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ services:
       JDK_JAVA_OPTIONS: -Xmx512M
       KAFKA_ADDRESS: kafka:9092
       SCHEMA_REGISTRY_URL: http://schemaregistry:8081
+      USAGE_REPORTS_FINGERPRINT: "e2e-tests"
```
Comment (dswiecki) [inline e2e-tests/docker-compose.yml:12]: isn't it better to disable usage statistics in tests at all ?

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/config/UsageStatisticsReportsConfig.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+private object FingerprintUtils {
+  lazy val random = s"gen-${Random.alphanumeric.take(10).mkString}"
+}
+
+case class UsageStatisticsReportsConfig(enabled: Boolean, fingerprint: String = FingerprintUtils.random)
+
+object UsageStatisticsUrl {
+  def apply(fingerprint: String, version: String) =
+    s"https://stats.nussknacker.io/?fingerprint=${URLEncoder.encode(fingerprint, StandardCharsets.UTF_8)}&version=${URLEncoder.encode(version, StandardCharsets.UTF_8)}&timestamp=${System.currentTimeMillis()}"
```
Comment (dswiecki) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/config/UsageStatisticsReportsConfig.scala:15]: what may pros and cons of having this url hardcoded vs hardcoded via configuration ?

Comment (gadomsky) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/config/UsageStatisticsReportsConfig.scala:15]: configuration can be overriden :)

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/WebResources.scala [hunk 3]
```diff
@@ -41,4 +43,10 @@ class WebResources(publicPath: String) extends Directives with LazyLogging {
       }
     }
   }
+
+  private def withUsageStatisticsReporting(html: String) = {
+    if(usageStatisticsReports.enabled) html.replace("</body>", s"""<img src="${UsageStatisticsUrl(usageStatisticsReports.fingerprint, BuildInfo.version)}" alt="" hidden /></body>""")
```
[the remark this answers is not in the document]
Comment (gadomsky) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/api/WebResources.scala]: added `alt` but `hidden` is a boolean parameter, so why?

File: e2e-tests/docker-compose.yml (hunk 1 above)
Comment (gadomsky) [reply e2e-tests/docker-compose.yml]: disabled

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/WebResources.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (mproch) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/api/WebResources.scala]: I'm somehow used to have html as proper xml (yes, yes, I know it's not 2010 anymore...)

**Review by mproch — APPROVED** [review]

**Pull request merged**